dotnet/src/Connectors/Connectors.GoogleVertexAI/GeminiPromptExecutionSettings.cs

```diff
@@ -0,0 +1,238 @@
+    {
+        get => this._maxTokens;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._maxTokens = value;
+        }
+    }
+
+    /// <summary>
+    /// The count of candidates. Possible values range from 1 to 8.
+    /// </summary>
+    [JsonPropertyName("candidate_count")]
+    public int? CandidateCount
+    {
+        get => this._candidateCount;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._candidateCount = value;
+        }
+    }
+
+    /// <summary>
+    /// Sequences where the completion will stop generating further tokens.
+    /// Maximum number of stop sequences is 5.
+    /// </summary>
+    [JsonPropertyName("stop_sequences")]
+    public IList<string>? StopSequences
+    {
+        get => this._stopSequences;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._stopSequences = value;
+        }
+    }
+
+    /// <summary>
+    /// Represents a list of safety settings.
+    /// </summary>
+    [JsonPropertyName("safety_settings")]
+    public IList<GeminiSafetySetting>? SafetySettings
+    {
+        get => this._safetySettings;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._safetySettings = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the behavior for how tool calls are handled.
+    /// </summary>
+    /// <remarks>
+    /// <list type="bullet">
+    /// <item>To disable all tool calling, set the property to null (the default).</item>
+    /// <item>
+    /// To allow the model to request one of any number of functions, set the property to an
+    /// instance returned from <see cref="ToolCallBehavior.EnableFunctions"/>, called with
+    /// a list of the functions available.
+    /// </item>
+    /// <item>
+    /// To allow the model to request one of any of the functions in the supplied <see cref="Kernel"/>,
+    /// set the property to <see cref="ToolCallBehavior.EnableKernelFunctions"/> if the client should simply
+    /// send the information about the functions and not handle the response in any special manner, or
+    /// <see cref="ToolCallBehavior.AutoInvokeKernelFunctions"/> if the client should attempt to automatically
+    /// invoke the function and send the result back to the service.
+    /// </item>
+    /// </list>
+    /// For all options where an instance is provided, auto-invoke behavior may be selected. If the service
+    /// sends a request for a function call, if auto-invoke has been requested, the client will attempt to
+    /// resolve that function from the functions available in the <see cref="Kernel"/>, and if found, rather
+    /// than returning the response back to the caller, it will handle the request automatically, invoking
+    /// the function, and sending back the result. The intermediate messages will be retained in the
+    /// <see cref="ChatHistory"/> if an instance was provided.
+    /// </remarks>
+    public ToolCallBehavior? ToolCallBehavior
```

Comment:
I don't think so. ToolCallBehavior is nested in GeminiPromptExecutionSettings class.

Reply:
> I don't think so. ToolCallBehavior is nested in GeminiPromptExecutionSettings class.

The property, yes. But not the type:
```C#
namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;

/// <summary>Represents a behavior for Gemini tool calls.</summary>
public abstract class ToolCallBehavior
```

Reply:
hmm what we should do with that? In theory I could move this type to GeminiPromptExecutionSettings class but then I have to change name of this type due to conflict with property...

Reply:
I'm not sure what to do here yet... probably worth a discussion...

Reply:
@Krzysztof318, lets use a prefix for the `ToolCallBehavior` as `GeminiToolCallBehavior` to avoid the clash, we might have this sorted out once the Function Call abstraction is in place.